dbms/src/AggregateFunctions/AggregateFunctionSumMap.h

```diff
@@ -111,7 +111,8 @@ class AggregateFunctionSumMapBase : public IAggregateFunctionDataHelper<
                 using IteratorType = typename MapType::iterator;
 
                 array_column.getData().get(values_vec_offset + i, value);
-                const auto & key = keys_vec.getElement(keys_vec_offset + i);
+                key_column.get(keys_vec_offset + i, key_field);
```

Comment:
This can compromise performance as we create temporary Field object.
As long as performance test is passed, it is affordable.